packages/microsite/src/cli/microsite-dev.tsx

```diff
@@ -296,19 +313,13 @@ export default async function dev(
         }
       };
 
-      const direct = await loadAndSSR(base);
-      if (direct) {
-        return next();
-      }
-      const index = await loadAndSSR(`${base}/index`);
-      if (index) {
+      const pagePath = await findPageComponentPathForBaseUrl(base);
+      if (!pagePath) {
         return next();
       }
-      const dynamic = await findPotentialMatch(base);
-      if (dynamic) {
-        await loadAndSSR(dynamic);
-      }
-      next();
+      const absoluteUrl = `/${base}`;
+      res.setHeader("Content-Type", "text/html");
+      res.end(await renderPage(pagePath, absoluteUrl));
```

Comment:
Perhaps you prefer these 3 lines to be extracted into a function (maybe called `ssrPage`?). I inlined it since the logic is straightforward but let me know what you think!

Reply:
I think it's fine inline! Doesn't bother me